server/src/main/java/org/opensearch/bootstrap/Security.java

```diff
@@ -41,9 +41,10 @@
 import org.opensearch.common.transport.PortsRange;
 import org.opensearch.env.Environment;
 import org.opensearch.http.HttpTransportSettings;
+import org.opensearch.javaagent.bootstrap.AgentPolicy;
 import org.opensearch.plugins.PluginInfo;
 import org.opensearch.plugins.PluginsService;
-import org.opensearch.secure_sm.SecureSM;
+import org.opensearch.secure_sm.policy.PolicyFile;
```

Comment:
we cannot use the [PolicyFile implementation in JDK21](https://github.com/openjdk/jdk21u/blob/master/src/java.base/share/classes/sun/security/provider/PolicyFile.java)? via ```--add-exports java.base/sun.security.provider=ALL-UNNAMED```

Reply:
> we cannot use the PolicyFile implementation in JDK21? via --add-exports java.base/sun.security.provider=ALL-UNNAMED

Yes, we can, but it is a temporary solution, I think the end game is to have OpenSearch 3 running on JDK-24

Reply:
>  In this case, we cannot run on JDK-24 but at least we won't require any breaking changes to add the support a bit later

@andrross @peterzhuamazon Is that acceptable?

Reply:
Agreed that it is a temporary solution but I would vote for picking this temporary solution and get to the end state of the Java agent sooner. That will help us to find any breaking changes in the plugin/s. 

We will continue to work on our own Policy parsing logic. A "fresh" logic may land next in 2-3 days weeks but it wont be necessarily tied to 3.0 and Java agent can be delivered without blockers or time-slips in 3.0. Whenever we have the policy parser ready that version we will swap the implementations and open gates for JDK-24.